src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -28,219 +86,1018 @@ const ActorInit En_Ik_InitVars = {
+    /* Deku bubble    */ DMG_ENTRY(1, DMG_EFF_HIT),
+    /* Deku launch    */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* UNK_DMG_0x12   */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* Zora barrier   */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* Normal shield  */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* Light ray      */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* Thrown object  */ DMG_ENTRY(1, DMG_EFF_HIT),
+    /* Zora punch     */ DMG_ENTRY(1, DMG_EFF_HIT),
+    /* Spin attack    */ DMG_ENTRY(1, DMG_EFF_HIT),
+    /* Sword beam     */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* Normal Roll    */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* UNK_DMG_0x1B   */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* UNK_DMG_0x1C   */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* Unblockable    */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* UNK_DMG_0x1E   */ DMG_ENTRY(0, DMG_EFF_HIT),
+    /* Powder Keg     */ DMG_ENTRY(1, DMG_EFF_HIT),
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit D_8092C160 = { 18, 25, 80, MASS_HEAVY };
+static CollisionCheckInfoInit sColChkInfoInit = { 18, 25, 80, MASS_HEAVY };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8092C168[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(targetArrowOffset, 2916, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 12, ICHAIN_CONTINUE),
     ICHAIN_F32_DIV1000(gravity, -1000, ICHAIN_STOP),
 };
 
-#endif
+static EffectBlureInit2 sBlureInit = {
+    0, 8, 0, { 255, 255, 150, 200 }, { 255, 255, 255, 64 }, { 255, 255, 150, 0 }, { 255, 255, 255, 0 }, 8,
+    0, 2, 0, { 0, 0, 0, 0 },         { 0, 0, 0, 0 },
+};
+
+void EnIk_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static s32 sDisplayListDesegmented = 0;
+    s32 i;
+    EnIk* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gIronKnuckleSkel, &gIronKnuckleWalkAnim, this->jointTable,
+                       this->morphTable, IRON_KNUCKLE_LIMB_MAX);
+    Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->colliderCylinder);
+    Collider_InitAndSetTris(globalCtx, &this->colliderTris, &this->actor, &sTrisInit, this->shieldColliderItems);
+    Collider_InitAndSetQuad(globalCtx, &this->colliderQuad, &this->actor, &sQuadInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTableArmor, &sColChkInfoInit);
+    this->actor.params = IK_GET_ARMOR_TYPE(&this->actor);
+    this->actor.params--;
+
+    Effect_Add(globalCtx, &this->effectIndex, EFFECT_BLURE2, 0, 0, &sBlureInit);
+    if (!sDisplayListDesegmented) {
+
+        for (i = 0; i < ARRAY_COUNT(sIronKnuckleArmorType); i++) {
+            sIronKnuckleArmorType[i][IK_TYPE_SILVER - 1] = Lib_SegmentedToVirtual(sIronKnuckleArmorType[i][0]);
+            sIronKnuckleArmorType[i][IK_TYPE_BLACK - 1] = Lib_SegmentedToVirtual(sIronKnuckleArmorType[i][1]);
+            sIronKnuckleArmorType[i][IK_TYPE_WHITE - 1] = Lib_SegmentedToVirtual(sIronKnuckleArmorType[i][2]);
```

Comment:
As I mentioned in discord, I don't think the color enum usage actually makes sense here. It should apply to the first dimension, but that is already covered with the loop iterator i. Instead the second dimension would just apply to something with the segments (`0x8 - 0xA`).